include/nbl/ext/CentralLimitBoxBlur/CBlurPerformer.h

```diff
@@ -0,0 +1,130 @@
+
+    typedef nbl_glsl_ext_Blur_Parameters_t Parameters_t;
+
+    struct DispatchInfo_t
+    {
+        uint32_t wg_count[3];
+    };
+
+    CBlurPerformer(video::ILogicalDevice* device, uint32_t maxDimensionSize, bool useHalfStorage);
+
+    static core::SRange<const asset::SPushConstantRange> getDefaultPushConstantRanges();
+
+    static core::smart_refctd_ptr<video::IGPUSpecializedShader> createSpecializedShader(const char* shaderIncludePath, const uint32_t axisDim, const bool useHalfStorage, video::ILogicalDevice* device);
+
+    //! Returns the required size of a buffer needed to hold the output of one pass.
+    //! Typically used as an intermediate storage.
+    //! For example, for a 2D blur, use a buffer of the reported size to hold the result of the 1st pass/dispatch --input to the second pass/dispatch.
+    static inline size_t getPassOutputBufferSize(const asset::VkExtent3D& dims, const uint32_t channelCount)
+    {
+        return dims.width * dims.height * dims.depth * channelCount * sizeof(float);
+    }
+
+    static inline uint32_t buildParameters(uint32_t numChannels, const asset::VkExtent3D& inputDimensions, Parameters_t* outParams, DispatchInfo_t* outInfos,
+        const float radius, const asset::ISampler::E_TEXTURE_CLAMP* wrappingType, const asset::ISampler::E_TEXTURE_BORDER_COLOR* borderColor = nullptr)
+    {
+        uint32_t passesRequired = 0u;
+        
+        if (numChannels)
+        {
+            for (uint32_t i = 0u; i < 3u; i++)
+            {
+                auto dim = (&inputDimensions.width)[i];
+                if (dim >= 2u)
+                    passesRequired++;
+            }
+        }
+
+        if (passesRequired)
+        {
+            for (uint32_t i = 0u; i < passesRequired; i++)
+            {
+                auto& params = outParams[i];
+
+                params.input_dimensions.x = inputDimensions.width;
+                params.input_dimensions.y = inputDimensions.height;
+                params.input_dimensions.z = inputDimensions.depth;
+        
+                assert(wrappingType[i] <= asset::ISampler::ETC_MIRROR);
+
+                const auto passAxis = i;
+                params.input_dimensions.w = (passAxis << 30) | (numChannels << 27) | (wrappingType[i] << 25);
+
+                if (borderColor)
+                {
+                    assert(borderColor[i] <= asset::ISampler::E_TEXTURE_BORDER_COLOR::ETBC_INT_OPAQUE_WHITE);
+                    params.input_dimensions.w |= borderColor[i] << 22;
+                }
+        
+                params.input_strides.x = 1u;
+                params.input_strides.y = inputDimensions.width;
+                params.input_strides.z = params.input_strides.y * inputDimensions.height;
+                params.input_strides.w = params.input_strides.z * inputDimensions.depth;
+
+                // At this point I'm wondering do you even need both input and output strides
+                params.output_strides = params.input_strides;
```

Comment:
I think the idea was to be able to transpose the array after every pass so that the stores are coalesced (match invocation ID increase with output memory address increase) just like we do in the Blit

If you can deduce the input strides from output strides somehow, great.

in reality we don't need 8 separate values, we probably only need 4:
- YZ, ZX, XY products
- XYZ full product